src/CustomElements/v1/CustomElements.js

```diff
@@ -0,0 +1,597 @@
+      }
+    },
+
+    /**
+     * Upgrades or customizes a custom element.
+     *
+     * @param {HTMLElement} element
+     * @param {CustomElementDefinition} definition
+     * @param {boolean} callConstructor
+     * @private
+     */
+    _upgradeElement: function(element, definition, callConstructor) {
+      var prototype = definition.constructor.prototype;
+      element.__proto__ = prototype;
+      if (callConstructor) {
+        this._setNewInstance(element);
+        element.__upgraded = true;
+        new (definition.constructor)();
+        console.assert(this._newInstance == null);
+      }
+
+      var observedAttributes = definition.observedAttributes;
+      if (definition.attributeChangedCallback && observedAttributes.length > 0) {
+        this._attributeObserver.observe(element, {
+          attributes: true,
+          attributeOldValue: true,
+          attributeFilter: observedAttributes,
+        });
+
+        // Trigger attributeChangedCallback for existing attributes.
+        // https://html.spec.whatwg.org/multipage/scripting.html#upgrades
+        for (var i = 0; i < observedAttributes.length; i++) {
+          var name = observedAttributes[i];
+          if (element.hasAttribute(name)) {
+            var value = element.getAttribute(name);
+            element.attributeChangedCallback(name, null, value);
+          }
+        }
+      }
+    },
+
+    /**
+     * @private
+     */
+    _handleAttributeChange: function(mutations) {
+      for (var i = 0; i < mutations.length; i++) {
+        var mutation = mutations[i];
+        if (mutation.type === 'attributes') {
+          var name = mutation.attributeName;
+          var oldValue = mutation.oldValue;
+          var target = mutation.target;
+          var newValue = target.getAttribute(name);
+          var namespace = mutation.attributeNamespace;
+          target['attributeChangedCallback'](name, oldValue, newValue, namespace);
+        }
+      }
+    },
+  }
+
+  // Closure Compiler Exports
+  window['CustomElementsRegistry'] = CustomElementsRegistry;
+  CustomElementsRegistry.prototype['define'] = CustomElementsRegistry.prototype.define;
+  CustomElementsRegistry.prototype['get'] = CustomElementsRegistry.prototype.get;
+  CustomElementsRegistry.prototype['whenDefined'] = CustomElementsRegistry.prototype.whenDefined;
+  CustomElementsRegistry.prototype['flush'] = CustomElementsRegistry.prototype.flush;
+  CustomElementsRegistry.prototype['polyfilled'] = CustomElementsRegistry.prototype.polyfilled;
+  CustomElementsRegistry.prototype['enableFlush'] = CustomElementsRegistry.prototype.enableFlush;
+
+  // patch window.HTMLElement
+
+  var origHTMLElement = win.HTMLElement;
+  win.HTMLElement = function HTMLElement() {
+    var customElements = win['customElements'];
+    if (customElements._newInstance) {
+      var i = customElements._newInstance;
+      customElements._newInstance = null;
+      return i;
+    }
+    if (this.constructor) {
```

Comment:
Suggest adding a comment here indicating that `_newInstance` is for upgrade and the other path is for creation.